### ISSUE TITLE
How to import entries with categories?

### ISSUE THREAD

**Issue by aaronbushnell** [boboldehampsink/import#75]
I have a CSV of entries that looks like this:

```
title, categories
My title, "Parent Category, Sub-category, Another Parent Category, Another Sub-Category"
```

But it is only importing the "Parent Category" and ignoring the sub-categories. I've also tried this format with no luck:

```
title, categories
My title, "parent-category/sub-category, another-parent-category/another-sub-category"
```

I'm probably missing something obvious here. Am I doing something wrong?

**Comment by boboldehampsink** [comment]
That last way should work. Verify if your slugs match exactly (so for instance, no parent-category-1 slug etc.)

**Comment by aaronbushnell** [comment]
Hm, no dice. This is what the category column of one row of my CSV looks like: 
```
everyday-wellness/play-and-exercise
```

And the "Everyday Wellness/Play and Exercise" category URL is
```
blog/everyday-wellness/play-and-exercise
```

I even tried changing the category column of my CSV to 
```
blog/everyday-wellness/play-and-exercise
```
and that didn't work either.

**Comment by aaronbushnell** [comment]
I should note that I am using the `develop` branch based off [this comment from a past issue](https://github.com/boboldehampsink/import/issues/23#issuecomment-66621067)

**Comment by aaronbushnell** [comment]
In case it helps, I've tried a couple other things to see if this would work. This is my category setup ![Category](http://cl.ly/image/0N0r0F1O422g/Screen%20Shot%202015-09-28%20at%205.44.31%20PM.png)

Here's the different options I tried in my Entries CSV:
```
Treatments,Cancer: Just shows Treatments on imported entry
Treatments/Cancer: No category selected on imported entry
/blog/treatments/cancer/: No category selected on imported entry
/blog/treatments/cancer: No category selected on imported entry
blog/treatments/cancer: No category selected on imported entry
/treatments/cancer/: No category selected on imported entry
treatments/cancer: No category selected on imported entry
```

**Comment by radarseven** [comment]
Hey @aaronbushnell,

I was having this issue too, but found a solution. 

The plugin is looking for matching `Categories` based on the category `URI`. Which means, if you don't have category URL's turned on for the category group (see attached screenshot) the plugin won't find any matches. Turning on category URL's for the group you are importing to should fix the issue. It did for me at least.

**Note:** The plugin tries to intelligently slugify the category title from the column in your import CSV file. So, as long as the column value from your CSV file when slugified is a match to your category `slug`, the category or categories should be imported.

In my case, I don't need category URL's, so I turned them off again once I finished importing.

Hope that helps!

Michael

<img width="618" alt="screenshot 2015-09-30 14 46 57" src="https://cloud.githubusercontent.com/assets/93583/10203182/7c4805e2-6782-11e5-86ab-569f77a05e26.png">

**Comment by boboldehampsink** [comment]
This is correct. Its described in the FAQ in the README aswel:

"How should I write my CSVs so parent & child entries/categories work properly?
Write it like Ancestor/Child/Entry and connect as "Ancestor". Do note that "Entries in this section have their own URLs" should be checked."

**Comment by boboldehampsink** [comment]
Did this fix your issue?

**Comment by aaronbushnell** [comment]
Hey, sorry about the delay, guys. My categories did have their own URLs so this unfortunately didn't resolve my issue.

However, I was able to fix it using a different strategy. I temporarily made my subcategories top-level categories. This correctly linked the posts to the right category. Then, once the import was complete, I moved my categories in the correct hierarchy again.

Odd, but it ended up working out!
